source/acquaman/AMContinuousScanActionController.cpp

```diff
@@ -207,3 +225,511 @@ void AMContinuousScanActionController::fillDataMaps(AMAgnosticDataAPIDataAvailab
+	for(int x = 0, size = interpolatedAxis_.count(); x < size; x++){
+		interpolatedAxis_[x] = currentAxisValue;
+		interpolatedMidpoints_[x] = currentAxisValue+(resolutionStep_/2);
+		currentAxisValue += resolutionStep_;
+	}
+
+	isUpScan_ = true;
+	if(axisFeedbackValues_.first() > axisFeedbackValues_.last())
+		isUpScan_ = false;
+
+
+	// Find the index mapping between interpolated points and feedback points
+	interpolatedMidpointsMappingIndices_ = QVector<double>(interpolatedSize_);
+	int currentInOrderEnergyLookupIndex = 0;
+	if(!isUpScan_)
+		currentInOrderEnergyLookupIndex = axisFeedbackValues_.count()-1;
+
+	double currentInOrderEnergyValue = axisFeedbackValues_.at(currentInOrderEnergyLookupIndex);
+	double lastInOrderEnergyValue = currentInOrderEnergyValue;
+	double fractionalIndex = 0;
+	double lastFractionalIndex = 0;
+
+	if(isUpScan_){
+		for(int x = 0, size = interpolatedSize_; x < size; x++){
+			double oneEnergyMidpoint = interpolatedMidpoints_.at(x);
+
+			while(oneEnergyMidpoint > currentInOrderEnergyValue){
+				currentInOrderEnergyLookupIndex++;
+				lastInOrderEnergyValue = currentInOrderEnergyValue;
+				currentInOrderEnergyValue = axisFeedbackValues_.at(currentInOrderEnergyLookupIndex);
+			}
+
+			lastFractionalIndex = fractionalIndex;
+			fractionalIndex = (currentInOrderEnergyLookupIndex-1) + ((oneEnergyMidpoint-lastInOrderEnergyValue)/(currentInOrderEnergyValue-lastInOrderEnergyValue));
+			interpolatedMidpointsMappingIndices_[x] = fractionalIndex;
+		}
+	}
+	else{
+		for(int x = 0, size = interpolatedSize_; x < size; x++){
+			double oneEnergyMidpoint = interpolatedMidpoints_.at(x);
+
+			while(oneEnergyMidpoint > currentInOrderEnergyValue){
+				currentInOrderEnergyLookupIndex--;
+				lastInOrderEnergyValue = currentInOrderEnergyValue;
+				currentInOrderEnergyValue = axisFeedbackValues_.at(currentInOrderEnergyLookupIndex);
+			}
+
+			lastFractionalIndex = fractionalIndex;
+			fractionalIndex = (currentInOrderEnergyLookupIndex+1) - ((oneEnergyMidpoint-lastInOrderEnergyValue)/(currentInOrderEnergyValue-lastInOrderEnergyValue));
+			interpolatedMidpointsMappingIndices_[x] = fractionalIndex;
+		}
+	}
+
+	return true;
+}
+
+bool AMContinuousScanActionController::generateInterpolatedScalerVectors()
+{
+	foreach(CLSAMDSScalerChannelDetector *scalerChannelDetector, scalerChannelDetectors_)
+		interpolatedScalerChannelVectors_.insert(scalerChannelDetector->name(), QVector<double>(interpolatedSize_));
+
+	// Loop over scaler interpolated vectors and fill them
+	QMap<QString, QVector<double> >::iterator b = interpolatedScalerChannelVectors_.begin();
+	while(b != interpolatedScalerChannelVectors_.end()){
+		QVector<qint32> oneOriginalVector = scalerChannelRebaseVectors_.value(b.key());
+
+		double startFractionalIndex;
+		double endFractionIndex;
+
+		if(isUpScan_){
+			int startFloorIndex;
+			int endFloorIndex;
+			for(int x = 0, size = b.value().count(); x < size; x++){
+				if(x == 0)
```

Comment:
Although a bit on the uglier side of things.  Doing the special cases of the first and last point outside the loop would minimize computation by removing those checks.